### DISCUSSION TITLE
Issue: Using "update()" with useFieldArray isn't working

### DISCUSSION THREAD

**Discussion by douglasrcjames** [react-hook-form/react-hook-form#10698] [Q&A] [answered]
### Version Number

Local project: "react-hook-form": "^7.44.3", Codepen: 7.4.1

### Codesandbox/Expo snack

https://codesandbox.io/s/happy-bush-9w5ytf?file=/src/App.tsx

### Steps to reproduce

When using the `update()` function for a `useFieldArray`, Typescript is giving an error that `update()` doesnt exist. Confused on the usage since I use to use the update function as below, but now on a new project this usage is giving a type error, however it seems to be working on my test environment! How do we use the update function to add/remove fields appropriately? Maybe I am using it wrong, but the docs seem to outline the update function as available for use, but on my codepen it looks like I am getting an error (that is different from my local project) that update isnt available!

<img width="626" alt="image" src="https://github.com/react-hook-form/react-hook-form/assets/22306305/67a534e6-0d4e-46e0-b255-9cc522b89570">

### Expected behaviour

Expected to be able to use `update()` function useFieldArray to change the field at said index as the docs outline!

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
On my local project:

Expected 2 arguments, but got 0.ts(2554)
fieldArray.d.ts(164, 155): An argument for 'index' was not provided.
```

On the attached codepen:
```
Property 'update' does not exist on type 'UseFieldArrayReturn<FormValues, "test", "id">'.
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment]
@douglasrcjames 

The root cause is from mismatch version of typescript inside your project and react-hook-form. Please upgrade to the latest react-hook-form to use with typescript version >= 5 or downgrade your project typescript version to 4.2.3 to stick with react-hook-form 7.4.1 .

**Reply by douglasrcjames** [reply]
Issue persists after updating react-hook-form to ^7.45.2 and Typescript is at ^5.1.6. Updated the codepen attached.
https://codesandbox.io/s/happy-bush-9w5ytf?file=/src/App.tsx

Also deleted node_modules and reinstalled. Reset Typescript server as well.

Any other suggestions?

**Reply by leapful** [reply]
@douglasrcjames 

The new codesandbox url is working fine on my side. Did you mean that the problem only persists on your local development ? If so, could you please help to check node_modules to see exact version of typescript ? Some projects might use resolution might to fix the version from package.json.

**Reply by douglasrcjames** [reply]
The code has been working yes, but Typescript is throwing an error, see below on the CodeSandbox with updated RHF and typescript versions, erroring on "update()" being called with no arguments (same as my local environment):
![image](https://github.com/react-hook-form/react-hook-form/assets/22306305/36bdc4cb-f6a9-401a-a566-5bcc75e8cc0d)

**Reply by leapful** [reply, marked as answer]
@douglasrcjames 

Method `update` from `useFieldArray` hook was never been called without parameters.

Reference: https://github.com/react-hook-form/react-hook-form/releases/tag/v7.11.0

<img width="1041" alt="Screenshot 2023-07-24 at 12 35 31" src="https://github.com/react-hook-form/react-hook-form/assets/18494222/4808bd57-98b9-40f5-a4f9-24ae2dc27e03">

**Reply by douglasrcjames** [reply]
Ahh yes, well these are pains from moving from JS to TS lol. I was using it previously with no params and it was working fine in JS! Thanks taking the time to clarify that with me 😄
